src/rules/selector-max-specificity/index.js

```diff
@@ -35,15 +35,21 @@ export default function (max) {
         resolvedNestedSelector(selector, rule).forEach(resolvedSelector => {
           // Return early if selector contains a not pseudo-class
           if (selector.indexOf(":not(") !== -1) { return }
+          // Return early if selector contains a matches
+          if (selector.indexOf(":matches(") !== -1) { return }
```

Comment:
@davidtheclark Presumably it'll also be more performant to return early if we know the `specificity` module is going to struggle to parse something?